src/c/native_ledger_client.cc

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2018 Brave Software. Distributed under the MPL2 license.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "src/c/native_ledger_client.h"
+
+#include "bat-native-ledger/include/bat/ledger/ledger.h"
+
+namespace native_ledger {
+  NativeLedgerClient::NativeLedgerClient() :
+    ledger_(ledger::Ledger::CreateInstance(this)) {
+  }
+
+  NativeLedgerClient::~NativeLedgerClient() {
+  }
+
+  void NativeLedgerClient::CreateWallet() {
+    ledger_->CreateWallet();
+  }
+
+  std::string NativeLedgerClient::GenerateGUID() const {
+    return "empty_string";
+  }
+
+  void NativeLedgerClient::Shutdown() {
+    ledger_.reset();
+  }
+
+  void NativeLedgerClient::OnWalletCreated(ledger::Result result) {
+    return;
+  }
+
+  void NativeLedgerClient::OnReconcileComplete
+    (ledger::Result result, const std::string& viewing_id) {
+    return;
+  }
+
+  void NativeLedgerClient::LoadLedgerState
+    (ledger::LedgerCallbackHandler* handler) {
+    handler->OnLedgerStateLoaded(ledger::Result::OK, ledger_state_);
```

Comment:
this is where you're going to want to hand off to JS. You can wrap the handler callback and give it to JS as well so it can notify Ledger when the state is loaded